### PR TITLE
Upgrade net45 to net462 and Castle.Core to 5.1.1

### PR DESCRIPTION
**Pull request by ViktorHofer**
Fixes https://github.com/FakeItEasy/FakeItEasy/issues/1925

- Upgrade net45 to net462 as net45 has been out-of-support for years and doesn't support Castle.Core 5.x
- Upgrade Castle.Core 4.3.1 to 5.1.1 as the former version doesn't provide a netstandard2.0 implementation which results in the FakeItEasy package bringing in the entire .NET Standard 1.5 dependency graph. See the linked issue for more details.
- Use TargetFrameworkIdentifier / the corresponding property function (for property groups) to condition on .NETFramework instead on a specific .NET Framework TFM.

### CODE REVIEW

**Comment by ViktorHofer** [comment]
@blairconrad @thomaslevesque looks like the `FakeItEasy.Build: spec/tests/FakeItEasy.Specs: Starting...` step is failing. Can you please take a look? I'm not that familiar with this library and it would probably take me way longer to diagnose than you. Thanks

**Comment by blairconrad** [comment]
> @blairconrad @thomaslevesque looks like the `FakeItEasy.Build: spec/tests/FakeItEasy.Specs: Starting...` step is failing. Can you please take a look? I'm not that familiar with this library and it would probably take me way longer to diagnose than you. Thanks

Yeah, these are caused by the differences in the exceptions that Castle.Core 5.1.1 throws when it can't create a proxy. I alluded to in https://github.com/FakeItEasy/FakeItEasy/issues/1925#issuecomment-1510346158. I've pushed a fixup commit to deal with them, as well as the Api Approval step that the original build didn't get to.

(And a bonus commit adding the profile files to the sln, as we should've had all along. Not strictly necessary for this PR, but since we were changing the file list anyhow, seemed like a good time.)

File: how_to_build.md
Comment (ViktorHofer) [reply how_to_build.md:35]: At the moment that's true, yes. When building from the CLI, the targeting pack is automatically restore and used but currently not inside VS, which is a tooling limitation. The good news is that we are actively working on fixing that with the next version of the .NET SDK and Visual Studio.

**Comment by ViktorHofer** [comment]
> DontMergeMeYet

That leg is interesting. I think you can achieve the same by disabling merge and rebase in the repository settings.

And as I'm already providing feedback on the infrastructure, you don't really need the profiles props files as the .NET SDK already supports this out of the box. Examples: `dotnet build -f net462`, `dotnet build -f netstandard2.0`.

**Comment by ViktorHofer** [comment]
Sounds good 👍

**Review by thomaslevesque — APPROVED** [review]
Looks good to me, thanks @ViktorHofer!

This is a breaking change, so we'll release in a a new major version. But we have other changes we'd like to make before bumping the major version, so we won't merge this immediately. (We don't really have a good process for non-linear development at the moment, because we usually don't need it)

**Comment by thomaslevesque** [comment]
Oh! Actually, I had forgotten that I had a pending comment requesting a change. Could you please address it, and squash the fixup commit?

File: how_to_build.md
Comment (ViktorHofer) [reply how_to_build.md:35]: Done

**Comment by ViktorHofer** [comment]
> Oh! Actually, I had forgotten that I had a pending comment requesting a change. Could you please address it, and squash the fixup commit?

Done :)

> This is a breaking change, so we'll release in a a new major version. But we have other changes we'd like to make before bumping the major version, so we won't merge this immediately. (We don't really have a good process for non-linear development at the moment, because we usually don't need it)

Fully understood. Would it be possible to share an expectation of when that will happen? Asking as I would like to take a dependency on the updated package.

**Comment by blairconrad** [comment]
> (We don't really have a good process for non-linear development at the moment, because we usually don't need it)

I think "not usually needing it" is a _fantastic_ process for non-linear development!

**Comment by blairconrad** [comment]
> > This is a breaking change, so we'll release in a a new major version. But we have other changes we'd like to make before bumping the major version, so we won't merge this immediately. […]
> 
> Fully understood. Would it be possible to share an expectation of when that will happen? Asking as I would like to take a dependency on the updated package.

I'm the one pushing to finish a few minor updates. I'm distracted by a broken washing machine right now, but I'd expect days, not weeks.

**Comment by ViktorHofer** [comment]
Thanks. All the best with your washing machine ;)

**Review by thomaslevesque — APPROVED** [review]

**Review by thomaslevesque — APPROVED** [review]

**Comment by thomaslevesque** [comment]
Thanks, @ViktorHofer!

**Comment by blairconrad** [comment]
Thanks, @ViktorHofer. Thanks @thomaslevesque for the merge!

**Comment by blairconrad** [comment]
> I would like to take a dependency on the updated package.

Oh, thanks. Folk came last Thursday to repair it. It cost approximately 1/4 of a replacement washing machine, and was less hassle than getting a new machine! (And less hassle than _me_ continuing to tinker.)

**Pull request merged**
